src/ds/ds_htable.c

```diff
@@ -249,6 +249,17 @@ static inline uint32_t get_resource_hash(zval *resource)
     return Z_RES_HANDLE_P(resource);
 }
 
+static inline uint32_t get_double_hash(zval *value)
+{
+    union ulld {
+        unsigned long long ull;
+        double d;
+    } hack;
+
+    hack.d = Z_DVAL_P(value);
+    return (uint32_t)(hack.ull ^ (hack.ull >> 32));
```

Comment:
Not sure how exactly ds defines floating point equality for the purpose of HT lookups, but most likely `0. === -0.`, which would not be handled correctly by this hash.

Reply:
Good catch - `0.0` and `-0.0` are mapped to the same value on `master`.

Reply:
@nikic `===` is used.